src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,128 @@
+import { useState } from 'react';
+
+type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
+
+type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user: User | null;
+};
+
+type Props = {
+  users: User[];
+  todoAdd: (newTodo: Todo) => void;
+  newTodoId: number;
+};
+
+export const TodoForm: React.FC<Props> = ({ users, todoAdd, newTodoId }) => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const titleRegex = /^[a-zA-Zа-яА-Я0-9\s]*$/;
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const getUserById = (userId: number): User | null => {
```

Comment:
- You have the same function in `App`
- 'userId' is already declared in the upper scope on line 28 column 10